File: parse_blogs.py
import json
import re
import os

def generate_slug(title):
    slug = title.lower()
    slug = re.sub(r'[^a-z0-9]+', '-', slug)
    return slug.strip('-')

def parse_blogs():
    with open('blogcontent.txt', 'r', encoding='utf-8') as f:
        content = f.read()

    sections = re.split(r'============BLOG \d+=============', content, flags=re.IGNORECASE)
    sections = [s for s in sections if s.strip()]

    blogs = []
    dates = ['April 26, 2026', 'April 20, 2026', 'April 15, 2026']

    for i, section in enumerate(sections):
        lines = [line.strip() for line in section.strip().split('\n') if line.strip()]
        
        meta_title = ''
        meta_description = ''
        title = ''
        body_start_index = 0

        for j, line in enumerate(lines):
            lower_line = line.lower()
            if lower_line.startswith('meta title'):
                meta_title = re.sub(r'^meta title\s*:?\s*', '', line, flags=re.IGNORECASE)
            elif lower_line.startswith('meta description'):
                meta_description = re.sub(r'^meta description\s*:?\s*', '', line, flags=re.IGNORECASE)
            else:
                title = line
                body_start_index = j + 1
                break

        paragraphs = lines[body_start_index:]

        summary = ''
        for p in paragraphs:
            if len(summary) < 150:
                summary += (' ' if summary else '') + p
            else:
                break
                
        if len(summary) > 150:
            summary = re.sub(r'\s+\S*$', '', summary[:147]) + '...'

        slug = generate_slug(title)

        blogs.append({
            'id': slug,
            'slug': slug,
            'metaTitle': meta_title,
            'metaDescription': meta_description,
            'title': title,
            'date': dates[i] if i < len(dates) else dates[0],
            'summary': summary,
            'content': paragraphs
        })

    target_dir = os.path.join('src', 'lib', 'data')
    os.makedirs(target_dir, exist_ok=True)

    with open(os.path.join(target_dir, 'blogs.json'), 'w', encoding='utf-8') as f:
        json.dump(blogs, f, indent=2)

    print(f"Successfully parsed {len(blogs)} blogs and saved to src/lib/data/blogs.json")

if __name__ == '__main__':
    parse_blogs()

File: parse_blogs.js
import fs from 'fs';
import path from 'path';
import crypto from 'crypto';

function generateSlug(title) {
  return title
    .toLowerCase()
    .replace(/[^a-z0-9]+/g, '-')
    .replace(/(^-|-$)+/g, '');
}

function parseBlogs() {
  const content = fs.readFileSync('blogcontent.txt', 'utf8');
  const sections = content.split(/=+\s*BLOG\s*\d+\s*=+/i).filter(s => s.trim().length > 0);
  
  const apiDir = path.join('static', 'api');
  if (!fs.existsSync(apiDir)) {
    fs.mkdirSync(apiDir, { recursive: true });
  }

  // Load existing blogs to preserve UUIDs
  const existingBlogsFile = path.join(apiDir, 'blogs.json');
  let existingBlogs = [];
  if (fs.existsSync(existingBlogsFile)) {
    try {
      existingBlogs = JSON.parse(fs.readFileSync(existingBlogsFile, 'utf8'));
    } catch (e) {
      console.error('Error parsing existing blogs.json', e);
    }
  }

  const blogsList = [];

  sections.forEach((section, index) => {
    const lines = section.trim().split('\n').map(line => line.trim()).filter(line => line.length > 0);
    
    let metaTitle = '';
    let metaDescription = '';
    let title = '';
    let bodyStartIndex = 0;
    
    for (let i = 0; i < lines.length; i++) {
      if (lines[i].toLowerCase().startsWith('meta title')) {
        metaTitle = lines[i].replace(/^meta title\s*:?\s*/i, '');
      } else if (lines[i].toLowerCase().startsWith('meta description')) {
        metaDescription = lines[i].replace(/^meta description\s*:?\s*/i, '');
      } else {
        title = lines[i];
        bodyStartIndex = i + 1;
        break;
      }
    }
    
    const paragraphs = lines.slice(bodyStartIndex);
    
    // Format to markdown
    let markdownContent = '';
    for (const paragraph of paragraphs) {
      if (paragraph.startsWith('Example')) {
        markdownContent += `### ${paragraph}\n\n`;
      } else if (/^\d+\./.test(paragraph)) {
        markdownContent += `### ${paragraph}\n\n`;
      } else if (paragraph.endsWith('?')) {
        markdownContent += `## ${paragraph}\n\n`;
      } else {
        markdownContent += `${paragraph}\n\n`;
      }
    }

    let summary = '';
    for (const p of paragraphs) {
      if (summary.length < 150) {
        summary += (summary.length > 0 ? ' ' : '') + p;
      } else {
        break;
      }
    }
    if (summary.length > 150) {
      summary = summary.substring(0, 147).replace(/\s+\S*$/, '') + '...';
    }
    
    const titleId = generateSlug(title);
    
    // Find existing UUID if it exists
    const existing = existingBlogs.find(b => b.titleId === titleId);
    const uuid = existing ? existing.id : crypto.randomUUID();
    
    const dates = [
      'April 26, 2026',
      'April 20, 2026',
      'April 15, 2026'
    ];
    const date = dates[index] || dates[0];
    
    // Add to lightweight list
    blogsList.push({
      id: uuid,
      titleId,
      title,
      date,
      summary,
      metaTitle,
      metaDescription
    });

    // Save full blog object to its own JSON file
    const fullBlogData = {
      id: uuid,
      titleId,
      metaTitle,
      metaDescription,
      title,
      date,
      summary,
      content: markdownContent.trim()
    };

    fs.writeFileSync(path.join(apiDir, `${uuid}.json`), JSON.stringify(fullBlogData, null, 2));
  });
  
  // Save the list
  fs.writeFileSync(path.join(apiDir, 'blogs.json'), JSON.stringify(blogsList, null, 2));
  console.log(`Successfully parsed ${blogsList.length} blogs to Markdown and saved to static/api/`);
}

parseBlogs();
